bool absorb
---
bool success
string message